altair/sphinxext/altairplot.py

```diff
@@ -216,13 +220,14 @@ def html_visit_altair_plot(self, node):
             chart = eval_block(node["code"], namespace)
         stdout = f.getvalue()
     except Exception as e:
-        warnings.warn(
-            "altair-plot: {}:{} Code Execution failed:"
-            "{}: {}".format(
-                node["rst_source"], node["rst_lineno"], e.__class__.__name__, str(e)
-            )
+        message = "altair-plot: {}:{} Code Execution failed:" "{}: {}".format(
+            node["rst_source"], node["rst_lineno"], e.__class__.__name__, str(e)
         )
-        raise nodes.SkipNode
+        if node["strict"]:
+            raise ValueError(message)
```

Comment:
Perhaps `raise ValueError(message) from e` to allow the error to include more context.